specs/www.rst

```diff
@@ -107,15 +109,16 @@ The *connection scope* information passed in ``scope`` contains:
   the iterable with ``host`` as the header name or replace any existing host
   header already present.
 
-* ``client`` (*Iterable[Unicode string, int]*) -- A two-item iterable
-  of ``[host, port]``, where ``host`` is the remote host's IPv4 or
-  IPv6 address, and ``port`` is the remote port as an
-  integer. Optional; if missing defaults to ``None``.
+* ``client`` (*Iterable[Unicode string, Optional[int]]*) -- A two-item
+  iterable of ``[host, port]``, where ``host`` is the remote host's
+  address, and ``port`` is the remote port as an integer or ``None``
+  if not appropriate. Optional; if missing defaults to ``None``.
```

Comment:
I presume this is meant to be written as "(host and port) or None", but it sort of reads like "host & (port or None)" - think we can tighten that up?

Reply:
I am expecting it to be `[host: str, port: int]` (IP) or `[host: str, None]` (Unix socket) or `None` (asgi-server can't represent/doesn't understand).  I'll reword to clarify this, if you are ok with this structure?

I think allowing port to be an `int` or `None` is the a minimal change that supports this without breaking frameworks. Although allowing the port to be an `int` or `-1` is probably the most minimal.

Reply:
Ah, I see! Then yes, let's definitely make these more robust.

Curious, though, what is your proposed value for `client host` in the UNIX socket case? Not sure what exactly would make sense there, the process ID is the closest thing I can think of and that's not really a string.

Reply:
I'm not sure about the client side, locally when I test unix sockets I can't get a peername. I think I should limit this to the server?

Reply:
Werkzeug's dev server uses `"<local>"`, but not sure how useful that is.

Reply:
Yeah, I would say `client` information isn't very useful when we have a unix socket and it should be left as an overall `None`; let's just focus on `server`.

Reply:
Thanks, I've updated.